### PR TITLE
[Fix](Variant) cast from jsonb with string value when enable `jsonb_string_as_string` should consider others types include date,datateime,array ...

### PR DESCRIPTION
**Pull request by eldenmoon**
This will correct result when meet json value like '2020-01-01'

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -719,14 +719,20 @@ struct ConvertImplGenericFromJsonb {
...
                     assert_cast<ColumnString&>(*col_to).insert_data(blob->getBlob(),
                                                                     blob->getBlobLen());
                     (*vec_null_map_to)[i] = 0;
                     continue;
                 }
-                std::string json_str = JsonbToJson::jsonb_to_json_string(val.data, val.size);
-                ReadBuffer read_buffer((char*)(json_str.data()), json_str.size());
+                std::string input_str;
+                if (context->jsonb_string_as_string() && value->isString()) {
+                    const auto* blob = static_cast<const JsonbBlobVal*>(value);
```
Comment (amorynan) [inline be/src/vec/functions/function_cast.h:730]: is here variant str just json binary value ?

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**